### PR TITLE
Actually print the full diff into the log when the log is too big

### PR DESCRIPTION
**Pull request by weswigham**
Otherwise, the "diff" we're so used to seeing reported on github hasn't actually been reported anywhere in the logs - only individual branch and main errors, not any comparisons between them.

### CODE REVIEW

**Comment by jakebailey** [comment]
To fix the CI just do `pnpm changeset --empty`; this doesn't need a version bump as the pipeline builds from source.

File: packages/dtslint-runner/src/post-results.ts [hunk 1]
```diff
@@ -46,7 +46,12 @@ async function main() {
       if (diffComment) {
         newComment += `There were interesting changes:\n`;
         if (newComment.length + diffComment.length + checkLogsMessage.length > 65535) {
-          newComment += `Changes are too big to display here, please check the log.`;
+          // hardlink directly into the output of this script
+          const detailedLogUrl = `https://typescript.visualstudio.com/TypeScript/_build/results?buildId=${buildId}&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=40b1ee41-44d6-5bba-aa04-4b76a5c732e5`;
```
Comment (jakebailey) [inline packages/dtslint-runner/src/post-results.ts:50]: What are the extra query params?

Comment (weswigham) [reply packages/dtslint-runner/src/post-results.ts:50]: Deep linking directly to the job and task within the job that refers to this script's output.

Comment (jakebailey) [inline packages/dtslint-runner/src/post-results.ts:50]: What are the extra query params?

Comment (jakebailey) [reply packages/dtslint-runner/src/post-results.ts:50]: Hm, do those still work if the pipeline changes (I want to move it to yaml at some point)

Comment (weswigham) [reply packages/dtslint-runner/src/post-results.ts:50]: The GUIDs will probably change if the underlying tasks shift.

Comment (weswigham) [reply packages/dtslint-runner/src/post-results.ts:50]: Notably, if the GUIDs don't exist in the job, it just links to the first job/task - it doesn't 404 or anything - so it's pretty safe to just use these for now and change them in the future when the underlying task changes.

Comment (jakebailey) [reply packages/dtslint-runner/src/post-results.ts:50]: Ah, that's not too bad then.

**Review by jakebailey — APPROVED** [review]

**Pull request merged**